### DISCUSSION TITLE
Night Vision / Flashlight

### DISCUSSION THREAD

**Discussion by Sikatsu** [Zrips/CMI#2766] [Feature Requests]
**Description of feature/improvement request:**

It would be nice if you can add a command like /cmi flashlight or /cmi nightvision.
This command will basically enable the nightvision effect for you.

- Milkbucket will not disable the flashlight.
- Night vision particles will not be shown.
- Turn on/off your own or someone else's flashlight.
- All messages can be customized.
- Remember your setting after server restart or server relog.

@Zrips What do you think?

**Comment by mercurialmusic** [comment]
Maybe something like `/cmi effect [playerName] night_vision [duration]`?

You can set up a /nightvision alias for this, or for more control perhaps set it up in a kit with a placeholder in the command for the [playerName], then alias the kit itself so running /nightvision would point to /cmi kit nightvision, and passing a player argument in the alias /nightvision [some player name] would come out as /cmi kit nightvision [some player name]. Not terribly difficult.

Turning it off again would be as simple as pointing something to the /cmi effect [playerName] clear command.

I'm not sure you'd be able to make it a toggle unless you can figure something fancy out with specialized commands, so perhaps make your aliases /flashlighton and /flashlightoff (e.g.).

Edit: There are also ways to apply the effect to an item which you could put on with /hat or something and it applies the effect to the player. That method would require more research.

**Comment by Va1a** [comment]
I second this, would be a pretty cool utility
